sls-versions/src/test/java/com/palantir/sls/versions/SlsVersionTests.java

```diff
@@ -76,4 +86,93 @@ public void testCheckWithNonOrderableVersion() {
     public void testCheckWithGarbage() {
         assertThat(SlsVersion.check("foo")).isFalse();
     }
+
+    @ParameterizedTest
+    @MethodSource("serializable_arguments")
+    public void serializable(int version, SlsVersion input, SlsVersion expected) throws Exception {
+        Path path =
+                SERIALIZABLE_CASES_PATH.resolve(String.format("v" + version)).resolve(input.getValue() + ".ser");
+
+        if (!Files.exists(path)) {
+            writeObject(path, input);
+        }
+
+        SlsVersion actual = readObject(path, SlsVersion.class);
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    private static List<Arguments> serializable_arguments() {
+        return List.of(
+                Arguments.of(
+                        0,
+                        SlsVersion.valueOf("1.2.3-4-gabcdef"),
+                        new OrderableSlsVersion.Builder()
+                                .value("1.2.3-4-gabcdef")
+                                .majorVersionNumber(1)
+                                .minorVersionNumber(2)
+                                .patchVersionNumber(3)
+                                .firstSequenceVersionNumber(4)
+                                .type(SlsVersionType.RELEASE_SNAPSHOT)
+                                .build()),
+                Arguments.of(
+                        0,
+                        SlsVersion.valueOf("1.2.3"),
+                        new OrderableSlsVersion.Builder()
+                                .value("1.2.3")
+                                .majorVersionNumber(1)
+                                .minorVersionNumber(2)
+                                .patchVersionNumber(3)
+                                .type(SlsVersionType.RELEASE)
+                                .build()),
```

Comment:
Aren't these two things equal in all cases here? I'm wondering whether we should simply use `SlsVersion.valueOf("1.2.3")` and skip the builder itself, considering this is what developers would use to generate the versions (in most cases)